gwcs/wcs.py

```diff
@@ -332,7 +388,502 @@ def _invert(self, *args, **kwargs):
+            if self.bounding_box is None:
+                x0 = np.ones(self.pixel_n_dim)
+            else:
+                x0 = np.mean(self.bounding_box, axis=-1)
+
+        if arg_dim == 0:
+            argsi = args
+
+            if nargs == 2 and self._approx_inverse is not None:
+                x0 = self._approx_inverse(*argsi)
+                if not np.all(np.isfinite(x0)):
+                    return [np.array(np.nan) for _ in range(nargs)]
+
+            result = optimize.root(f, x0, **kwargs)
+            if result['success']:
+                return list(map(np.array, result['x']))
+            else:
+                return [np.array(np.nan) for _ in range(nargs)]
+
+        else:
+            arg_shape = args_shape[1:]
+            nelem = np.prod(arg_shape)
+            inv = np.empty((nargs, nelem))
+
+            args = np.reshape(args, (nargs, nelem))
+
+            failed_result = np.full(nargs, np.nan)
+
+            for k in range(nelem):
+                argsi = args[:, k]
+
+                if nargs == 2 and self._approx_inverse is not None:
+                    x0 = self._approx_inverse(*argsi)
+                    if not np.all(np.isfinite(x0)):
+                        inv[:, k] = failed_result
+                        continue
+
+                result = optimize.root(f, x0, **kwargs)
+
+                if result['success']:
+                    inv[:, k] = result['x']
+                else:
+                    inv[:, k] = failed_result
+
+            return list(np.reshape(inv, args_shape))
+
+    def _invert_scipy_fp(self, *args, **kwargs):
+        """
+        Implement iterative inverse here.
+        """
+        try:
+            from scipy import optimize
+        except ImportError:
+            raise ImportError("Computation of the iterative inverse requires 'scipy'")
+
+        # remove optional 'invert'-specific keywords:
+        kwargs = dict(kwargs)
+        kwargs.pop('with_bounding_box', 0)
+        kwargs.pop('fill_value', 0)
+
+        args_shape = np.shape(args)
+        nargs = args_shape[0]
+        arg_dim = len(args_shape) - 1
+
+        if nargs != self.world_n_dim:
+            raise ValueError("Number of input coordinates is different from "
+                             "the number of defined world coordinates in the "
+                             f"WCS ({self.world_n_dim:d})")
+
+        if self.world_n_dim != self.pixel_n_dim:
+            raise NotImplementedError(
+                "Support for iterative inverse for transformations with "
+                "different number of inputs and outputs was not implemented."
+            )
+
+        # estimate pixel scale using approximate algorithm
+        # from https://trs.jpl.nasa.gov/handle/2014/40409
+        if self.bounding_box is None:
+            crpix = np.ones(self.pixel_n_dim)
+        else:
+            crpix = np.mean(self.bounding_box, axis=-1)
+
+        l1, phi1 = np.deg2rad(self.__call__(*(crpix - 0.5)))
```

Comment:
Not really. To see what is going on, set `inv_pscale = 1` a few lines below and run the code on a large number of randomly distributed points in a large image.

Reply:
I'm not talking about completely random values for inv_pscale but why it is necessary to base it on the value at the reference pixel instead of based on the bounding box. If there is a large difference we have an enormous distortion.